### ISSUE TITLE
File include/exclude patterns

### ISSUE THREAD

**Issue by Nicoretti** [mpeterv/luacheck#21]
# GLOB patterns for including/excluding files
It would be nice if the user would be able to specify glob like include and exclude patterns for files which shall be checked (on the command line and the config file). 

# Reasoning
Some workspaces might define a .luacheckrc in the root directory. Usualy luacheck is used from there 

     user@host: ~/workspace$ luacheck . 

The problem is that the workspace also contains a tests directory and a directory for third party modules whose analysis aren't relevant. Even if the warnings of those files are suppressed/ignored they still add quite some noise (irrelevant data) to the output.

    Checking tests/_test1.lua          OK
    Checking tests/_tests_dir_includes.lua            OK
    Checking libs/cool_module1.lua            OK
    Checking libs/cool_module2.lua            OK
    Checking an_actually_important_file.lua            OK
    ...

**Comment by mpeterv** [comment]
Hello! On the command line the user could use wildcards (e.g. `luacheck *.lua` to only check files in current directory non-recursively), although IIRC globbing isn't supported in Windows shell. There can be several directories and files/wildcards as arguments, e.g. `luacheck *.lua src utils` will check files in the current directory non-recursively and files in `src` and `utils` recursively.

Of course that gets repetitive. I think a feature supplementing what you are suggesting would be a default list of files to check, probably specified using `files` option in config:

```lua
files = {"file1.lua", "files2.lua", "some_directory"}
-- other config options...
```

Filtering by filenames would still be useful e.g. if the user wants to focus on a certain part of the codebase. So I will look into implementing this for the next release. Do you have any suggestions on how to name the new options?

**Comment by Nicoretti** [comment]
Appriciate it! You might wanna use the same option names as the busted testing framework does.

    --filter=PATTERN      only run test names matching the Lua pattern
    --filter-out=PATTERN  do not run test names matching the Lua pattern

**Comment by thibaultCha** [comment]
That would be a very nice feature. Currently I have a project I want to lint, but one of the files is a fixture for my tests and it is a syntactically incorrect Lua file. Luacheck always throws an error at it and there is no way to ignore it.

**Comment by mpeterv** [comment]
I'll push a simple implementation this or next week but won't be able to release yet as #20 is only partially implemented right now. You could build file list for luacheck using `find` for now, e.g. `find src -name '*.lua' ! -name 'bad_file.lua' | xargs luacheck`.

**Comment by niuys** [comment]
you can change the log option to disable the log output.Maybe we need a new log option to only print error logs, it's noisy to see too many "ok" logs when the file num is a lot.

**Comment by mpeterv** [comment]
@niuys if you want to disable OK messages you can use `-q` flag.

**Comment by mpeterv** [comment]
So far `--exclude-files` option is implemented on master branch, it unterstands globbing, including recursive globbing. E.g. `find src -name '*.lua' ! -name 'bad_file.lua' | xargs luacheck` becomes `luacheck src --exclude-files '**/bad_file.lua'`. It can also be used in config.

**Comment by mpeterv** [comment]
Also added `--include-files` option. @Nicoretti @thibaultCha can this be closed once 0.11.0 is released?

**Comment by thibaultCha** [comment]
Yeah sure that's really great, thank you!
